doc/subframes/src/subframes_tutorial.cpp

```diff
@@ -0,0 +1,325 @@
+  box.subframe_poses[0].position.z = 0.0 + z_offset_box;
+  tf2::Quaternion orientation;
+  orientation.setRPY((90.0 / 180.0 * M_PI), 0, 0);
+  box.subframe_poses[0].orientation = tf2::toMsg(orientation);
+
+  box.subframe_names[1] = "top";
+  box.subframe_poses[1].position.y = .05;
+  box.subframe_poses[1].position.z = 0.0 + z_offset_box;
+  orientation.setRPY(-(90.0 / 180.0 * M_PI), 0, 0);
+  box.subframe_poses[1].orientation = tf2::toMsg(orientation);
+
+  box.subframe_names[2] = "corner_1";
+  box.subframe_poses[2].position.x = -.025;
+  box.subframe_poses[2].position.y = -.05;
+  box.subframe_poses[2].position.z = -.01 + z_offset_box;
+  orientation.setRPY((90.0 / 180.0 * M_PI), 0, 0);
+  box.subframe_poses[2].orientation = tf2::toMsg(orientation);
+
+  box.subframe_names[3] = "corner_2";
+  box.subframe_poses[3].position.x = .025;
+  box.subframe_poses[3].position.y = -.05;
+  box.subframe_poses[3].position.z = -.01 + z_offset_box;
+  orientation.setRPY((90.0 / 180.0 * M_PI), 0, 0);
+  box.subframe_poses[3].orientation = tf2::toMsg(orientation);
+
+  box.subframe_names[4] = "side";
+  box.subframe_poses[4].position.x = .0;
+  box.subframe_poses[4].position.y = .0;
+  box.subframe_poses[4].position.z = -.01 + z_offset_box;
+  orientation.setRPY(0, (180.0 / 180.0 * M_PI), 0);
+  box.subframe_poses[4].orientation = tf2::toMsg(orientation);
+
+  moveit_msgs::CollisionObject cylinder;
+  // Spawning on the fingers causes collisions, and it's not easy to allow the collisions
+  // through the planning scene interface. TODO(felixvd)
+  double z_offset_cylinder = .12;
+  cylinder.header.frame_id = "panda_hand";
+  cylinder.id = "cylinder";
+  cylinder.primitives.resize(1);
+  cylinder.primitive_poses.resize(1);
+  cylinder.primitives[0].type = box.primitives[0].CYLINDER;
+  cylinder.primitives[0].dimensions.resize(2);
+  cylinder.primitives[0].dimensions[0] = 0.06;   // height (along x)
+  cylinder.primitives[0].dimensions[1] = 0.005;  // radius
+  cylinder.primitive_poses[0].position.x = 0.0;
+  cylinder.primitive_poses[0].position.y = 0.0;
+  cylinder.primitive_poses[0].position.z = 0.0 + z_offset_cylinder;
+  orientation.setRPY(0, (90.0 / 180.0 * M_PI), 0);
+  cylinder.primitive_poses[0].orientation = tf2::toMsg(orientation);
+
+  cylinder.subframe_poses.resize(1);
+  cylinder.subframe_names.resize(1);
+  cylinder.subframe_names[0] = "tip";
+  cylinder.subframe_poses[0].position.x = 0.03;
+  cylinder.subframe_poses[0].position.y = 0.0;
+  cylinder.subframe_poses[0].position.z = 0.0 + z_offset_cylinder;
+  orientation.setRPY(0, (90.0 / 180.0 * M_PI), 0);
+  cylinder.subframe_poses[0].orientation = tf2::toMsg(orientation);
+
+  // Publish each object
+  moveit_msgs::CollisionObject co;
+  co = box;
+  co.operation = moveit_msgs::CollisionObject::ADD;
+  planning_scene_interface.applyCollisionObject(co);
+  co = cylinder;
+  co.operation = moveit_msgs::CollisionObject::ADD;
+  planning_scene_interface.applyCollisionObject(co);
```

Comment:
Please use `applyCollisionObjects` here to perform the updates in one ROS service call.